connector/src/main/java/com/datastax/oss/pulsar/source/CassandraSource.java

```diff
@@ -343,7 +349,9 @@ void initCassandraClient() throws InvocationTargetException, NoSuchMethodExcepti
     synchronized void setValueConverterAndQuery(KeyspaceMetadata ksm, TableMetadata tableMetadata) {
         try {
             List<ColumnMetadata> columns = tableMetadata.getColumns().values().stream()
-                    .filter(c -> !tableMetadata.getPrimaryKey().contains(c))
+                    // include primary keys in the json only output format options
+                    // TODO: PERF: Infuse the key values returned from the mutation keys instead of reading from DB
```

Comment:
what about creating a GH issue and add a link here ?

Reply:
added in the new patch: https://github.com/datastax/cdc-apache-cassandra/issues/84